src/game_frontend/src/hooks/useWallet.tsx

```diff
@@ -0,0 +1,30 @@
+import { useAuth } from "./useAuth";
+
+const useWallet = (): string | undefined => {
+  const auth = useAuth();
+
+
+
+  // Determine the current authenticated wallet
+  const currentWallet: WalletName | undefined = (() => {
```

Comment:
Look into `useMemo` or `useCallback` and see which one will be more optimized solution for this function

Reply:
Going with ```useMemo```